aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -98,13 +100,31 @@ impl<'r> TStateView for ChangeSetStateView<'r> {
     }
 
     fn get_state_value(&self, state_key: &Self::Key) -> Result<Option<StateValue>> {
-        match self.change_set.delta_change_set().get(state_key) {
+        // TODO: `get_state_value` should differentiate between different types
+        // of storage data.
+        if matches!(state_key.inner(), StateKeyInner::AccessPath(ap) if ap.is_code()) {
+            // We are fetching a module.
+            return match self.change_set.module_write_set().get(state_key) {
+                Some(write_op) => Ok(write_op.as_state_value()),
+                None => self.base.get_state_value(state_key),
+            };
+        }
+
+        match self.change_set.aggregator_delta_set().get(state_key) {
+            // The value to resolve delta has to come from `base`.
             Some(delta_op) => Ok(delta_op
                 .try_into_write_op(self.base, state_key)?
                 .as_state_value()),
-            None => match self.change_set.write_set().get(state_key) {
-                Some(write_op) => Ok(write_op.as_state_value()),
-                None => self.base.get_state_value(state_key),
+            None => {
+                // Otherwise, this can be either a resource or an aggregator write.
+                match self.change_set.resource_write_set().get(state_key) {
```

Comment:
this part is ugly, should we just have a get function for the change_set to check all the write ops?

Reply:
Yes! I changed it to use write set iterator + find() for now